### DISCUSSION TITLE
std::ifstream pointer position not reset-ed after being used by json::accept(file);

### DISCUSSION THREAD

**Discussion by kenzanin** [nlohmann/json#4283] [Q&A] [answered]
code:
```cpp
    std::ifstream ifs(filename.data(), std::ifstream::in);
    if (!json::accept(ifs)) {
      log->critical("error file {} not valid config.", filename);
      std::exit(EXIT_FAILURE);
    }
    js = nlohmann::json::parse(ifs);
    log->debug("{}", js.dump());
```
expected:
js contains json object

actual:
terminate called after throwing an instance of 'nlohmann::json_abi_v3_11_2::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal

working:
```cpp
    std::ifstream ifs(filename.data(), std::ifstream::in);
    if (!json::accept(ifs)) {
      log->critical("error file {} not valid config.", filename);
      std::exit(EXIT_FAILURE);
    }
    ifs.seekg(0); // add this
    js = nlohmann::json::parse(ifs);
    log->debug("{}", js.dump());

```
question:
is this intentional? 

thanks

**Comment by nlohmann** [comment, marked as answer]
Yes, this is intentional:

- You need to reset your stream if you want to read it again.
- The library does not mess with the stream position and leaves the stream past the last byte it read. This allows use cases like https://json.nlohmann.me/features/parsing/json_lines/

**Reply by kenzanin** [reply]
thanks
